src/node.rs

```diff
@@ -573,33 +573,65 @@ impl EventStepper for Node {
 impl Node {
     /// Returns the chain for this node.
     pub fn chain(&self) -> Option<&Chain> {
-        self.machine.chain()
+        self.machine.current().chain()
+    }
+
+    /// Returns this node state.
+    pub fn node_state(&self) -> Option<&crate::states::Node> {
+        match *self.machine.current() {
+            State::Node(ref state) => Some(state),
+            _ => None,
+        }
+    }
+
+    /// Returns this node mut state.
+    pub fn mut_node_state(&mut self) -> Option<&mut crate::states::Node> {
```

Comment:
This should be called `node_state_mut` to follow the convention that is common in rust and its ecosystem.